core/src/player.rs

```diff
@@ -173,8 +173,17 @@ pub struct Player {
     background_color: Color,
 
     frame_rate: f64,
+
+    /// A time budget for executing frames.
+    /// Gained by passage of time between host frames, spent by executing SWF frames.
+    /// This is how we support custom SWF framerates
+    /// and compensate for small lags by "catching up" (up to MAX_FRAMES_PER_TICK).
     frame_accumulator: f64,
 
+    /// Faked time passage for fooling hand-written busy-loop FPS limiters.
+    /// All script-visible time sources must be delayed by this amount of ms.
+    time_offset: u32,
```

Comment:
Can we change this to a Duration? It's not clear that it means milliseconds, and makes interacting with it easier and more understandable

Reply:
> and makes interacting with it easier

I tried doing this, and... for now it's actually messier and harder :/ As we have a weird mix of floats, time::Duration and chrono::Duration across the codebase, and especially the latter two are awkward to mix.
And technically this is currently consistent with `frame_accumulator` which is also implicitly a number of milliseconds. So I think this should rather get refactored all at the same time, in a separate PR.